src/scripts/achievements/AchievementHandler.ts

```diff
@@ -19,12 +21,26 @@ class AchievementHandler {
     }
 
     public static getNumberOfTabs() {
-        return Math.floor(AchievementHandler.achievementList.filter(a => a.region <= player.highestRegion()).length / 10);
+        this.numberOfTabs(Math.floor(this.achievementListFiltered.filter(a => a.region <= player.highestRegion()).length / 10));
     }
 
-    public static getAchievementListWithIndex(index: number) {
-        index *= 10;
-        return AchievementHandler.achievementList.filter(a => a.region <= player.highestRegion()).slice(index, index + 10);
+    public static filter = {
+        status: ko.observable('all'),
+        type: ko.observable('all'),
+    }
+
+    public static getAchievementListWithIndex() {
+        this.achievementListFiltered = this.achievementList.filter(a => a.region <= player.highestRegion() &&
+                                        (this.filter.status() == 'all' ? true : a.unlocked == JSON.parse(this.filter.status())) &&
+                                        (this.filter.type() == 'all' ? true : a.type == achievementType[this.filter.type()]));
+        this.getNumberOfTabs();
+        return this.achievementListFiltered.slice(this.navigateIndex() * 10, (this.navigateIndex() * 10) + 10);
+    }
+
+    public static getAchievementOption(status: string, type: string) {
```

Comment:
`get` implies that this function is for retrieving a value, but this doesn't return anything.
I think it makes more sense to call this `updateAchievementOption` or even `updateAchievementFilter`

Reply:
you are right, this function is weird, I will dismember it to each filter and change its name, what do you know?

Reply:
That sounds like a good idea to me